src/modules/ipfix/aggregator/Rules.cpp

```diff
@@ -133,6 +133,26 @@ int parseProtoPattern(char* s, IpfixRecord::Data** fdata, InformationElement::Ie
 	return 0;
 }
 
+/**
+ * parses the given string
+ * @return 0 if successful
+ */
+int parseMacAddressPattern(char* s, IpfixRecord::Data** fdata, InformationElement::IeLength* length) {
+	unsigned char a[6];
+	int last = -1;
+	int rc = sscanf(s, "%hhx:%hhx:%hhx:%hhx:%hhx:%hhx%n",
```

Comment:
I suggest to use `get_next_token()` and iterate over `s` to make the code more readable.